src/gui/layout/qgslayoutviewtooladdnodeitem.cpp

```diff
@@ -69,15 +69,22 @@ void QgsLayoutViewToolAddNodeItem::layoutPressEvent( QgsLayoutViewMouseEvent *ev
     // last (temporary) point is removed
     mPolygon.remove( mPolygon.count() - 1 );
 
-    QgsLayoutItem *item = QgsGui::layoutItemGuiRegistry()->createItem( mItemMetadataId, layout() );
+    std::unique_ptr< QgsLayoutItem > item( QgsGui::layoutItemGuiRegistry()->createItem( mItemMetadataId, layout() ) );
     if ( !item )
       return;
 
-    if ( QgsLayoutNodesItem *nodesItem = qobject_cast< QgsLayoutNodesItem * >( item ) )
+    if ( QgsLayoutNodesItem *nodesItem = qobject_cast< QgsLayoutNodesItem * >( item.get() ) )
+    {
       nodesItem->setNodes( mPolygon );
+      if ( !nodesItem->isValid() )
+      {
+        mRubberBand.reset();
+        return;
+      }
+    }
 
-    layout()->addLayoutItem( item );
-    layout()->setSelectedItem( item );
+    layout()->addLayoutItem( item.get() );
+    layout()->setSelectedItem( item.release() );
```

Comment:
I get the rationale here, but I'm slightly -1 toward abusing the ownership of the pointer in this way (just in case we ever get to a stage where we use unique_ptrs to handle ownership explicitly in the api, this kind of thing will get messy). I'd just go with:

```suggestion
    QgsLayoutItem* newItem = item.get();
    layout()->addLayoutItem( item.release() );
    layout()->setSelectedItem( newItem );
```